lib/github-interface.js

```diff
@@ -0,0 +1,180 @@
+const rc = require('rc')
+const request = require('request-promise-native')
+
+const config = rc('github', { username: '', token: '' })
+const auth = {
+  user: config.username,
+  pass: config.token
+}
+
+const githubApiBase = 'https://api.github.com'
+const githubHeaders = {
+  'User-Agent': 'tableflip-adventure'
+}
+
+// Aggregates stats for the given release of the supplied repos.
+// Returns a promise which resolves to an object of the form:
+//
+// {
+//   commits: 100,
+//   additions: 1234,
+//   deletions: 123
+// }
+//
+// For example:
+//
+// releaseStats({
+//   repos: ['tableflip/project-frontend', 'tableflip/project-api'],
+//   version: 'v2.0.0'
+// })
+function releaseStats ({ repos, version }) {
+  const stats = repos.map((repo) => repoReleaseStats({ repo, version }))
+  return Promise.all(stats)
+    .then((reposData) => {
+      return reposData.reduce((memo, repoData) => {
+        return {
+          commits: memo.commits + repoData.commits,
+          additions: memo.additions + repoData.additions,
+          deletions: memo.deletions + repoData.deletions
+        }
+      }, { commits: 0, additions: 0, deletions: 0 })
+    })
+}
+
+// Gets the aggregate stats for the given repo and version, by comparing it to the previous tagged version if available,
+// or summarising the whole history if not
+async function repoReleaseStats ({ repo, version }) {
+  const releaseTags = await getTags({ repo })
+  const tagIndex = releaseTags.indexOf(version)
+
+  if (tagIndex === -1) throw new Error(`There is no release tagged ${version}`)
+
+  const previousVersion = releaseTags[tagIndex + 1]
+
+  if (previousVersion) {
+    return getCommitComparison({ repo, lastCommit: version, firstCommit: previousVersion })
+  } else {
+    return getRepoStats({ repo })
+  }
+}
+
+// Gets aggregate comparison stats between the two supplied commits (which could be branches or tags)
+async function getCommitComparison ({ repo, firstCommit, lastCommit }) {
+  const stats = await request({
+    method: 'GET',
+    url: `${githubApiBase}/repos/${repo}/compare/${firstCommit}...${lastCommit}`,
+    auth,
+    json: true,
+    headers: githubHeaders
+  })
+
+  const lineStats = stats.files.reduce((memo, file) => {
+    return {
+      additions: memo.additions + file.additions,
+      deletions: memo.deletions + file.deletions
+    }
+  }, { additions: 0, deletions: 0 })
+
+  return {
+    commits: stats.commits.length,
+    additions: lineStats.additions,
+    deletions: lineStats.deletions
+  }
```

Comment:
Just comparing this with the [pulse page](https://github.com/tableflip/study-bundles-app/pulse), can we return the number of files changed, PRs merged and issues closed?